client/components/app.jsx

```diff
@@ -35,16 +38,47 @@ export default class App extends React.Component {
       case 'catalog' :
         return <ProductList setView={this.setView} />;
       case 'details' :
-        return <ProductDetails setView={this.setView} params={this.state.view.params}/>;
+        return <ProductDetails setView={this.setView} params={this.state.view.params} addToCart={this.addToCart}/>;
       default :
         return null;
     }
   }
 
+  getCartItems() {
+    fetch('/api/cart')
+      .then(response => {
+        return response.json();
+      })
+      .then(data => {
+        this.setState({
+          cart: data
+        });
+      })
+      .catch(err => console.error(err));
+  }
+
+  addToCart(product) {
+    fetch('/api/cart', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json'
+      },
+      body: JSON.stringify(product)
+    })
+      .then(response => response.json())
+      .then(cartItem => {
+        const newCartItems = [...this.state.cart];
+        newCartItems.push(cartItem);
+        this.setState({
+          cart: newCartItems
+        }, () => this.getCartItems);
```

Comment:
You can leave out this extra callback, since the cart state is arguably up-to-date.